### PR TITLE
New way of setting audit logging tenant

### PR DESCRIPTION
**Pull request by Maladie**
**Description**

Changes proposed in this pull request:

- setting audit log tenant is now based on seed region

**Related issue(s)**

### CODE REVIEW

File: components/provisioner/internal/gardener/auditlogging.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+func getAuditLogInstanceIdentifier(seed gardener_types.Seed) string {
+	message := findAuditLogConditionMessage(seed)
+
+	if message == "" {
+		return ""
+	}
+
+	pat := regexp.MustCompile(audiInstanceCodePattern)
```
Comment (franpog859) [inline components/provisioner/internal/gardener/auditlogging.go:94]: Maybe we could compile the regexp in the `NewAuditLogConfigurator` function just to do it once and not every time we want to use it

File: components/provisioner/internal/gardener/auditlogging.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+
+	gardener_types "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+const auditLogConditionType = "AuditlogServiceAvailability"
```
Comment (franpog859) [inline components/provisioner/internal/gardener/auditlogging.go:13]: Did you look at the Gardener packages for this constant? If there is no way to get it through a package then it's ok, but if we can make it not hardcoded it'd be much better

File: components/provisioner/internal/gardener/auditlogging_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+		//when
+		enable := auditLogConfigurator.CanEnableAuditLogsForShoot(seedName)
+
+		//then
+		assert.False(t, enable)
+	})
+}
+
+func TestAuditLogConfigurator_SetAuditLogAnnotation(t *testing.T) {
```
Comment (franpog859) [inline components/provisioner/internal/gardener/auditlogging_test.go:51]: There could be also a test when the shoot is already annotated so the function will return `false, nil`, right?

File: components/provisioner/internal/gardener/shoot_reconciler.go [hunk 4]
```diff
@@ -98,17 +95,9 @@ func (r *Reconciler) shouldReconcileShoot(shoot gardener_types.Shoot) (bool, err
...
-		var refetchedShoot gardener_types.Shoot
-		err := r.client.Get(context.Background(), namespacedName, &refetchedShoot)
-		if err != nil {
-			return err
-		}
-
-		modifyShootFn(&refetchedShoot)
-
-		err = r.client.Update(context.Background(), &refetchedShoot)
+		err := r.client.Update(context.Background(), modifiedShoot)
```
Comment (franpog859) [inline components/provisioner/internal/gardener/shoot_reconciler.go:100]:
```suggestion
		return r.client.Update(context.Background(), modifiedShoot)
```
The error checking below is redundant here

File: components/provisioner/internal/gardener/shoot_reconciler.go [hunk 5]
```diff
@@ -72,11 +68,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
-	seed := getSeed(shoot)
-	if seed != "" && r.auditLogTenantConfigPath != "" {
-		err := r.enableAuditLogs(log, req.NamespacedName, &shoot, seed)
+	seedName := getSeedName(shoot)
+
+	if r.auditLogConfigurator.CanEnableAuditLogsForShoot(seedName) {
+		err := r.enableAuditLogs(log, &shoot, seedName)
 		if err != nil {
-			log.Errorf("Failed to enable audit logs for %s shoot: %s", shoot.Name, err.Error())
+			log.Errorf("failed to enable audit logs for %s shoot: %s", shoot.Name, err.Error())
```
Comment (franpog859) [inline components/provisioner/internal/gardener/shoot_reconciler.go:76]: Why though :D

File: components/provisioner/internal/gardener/auditlogging.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+		auditLogTenantConfigPath: auditLogTenantConfigPath,
+	}
+}
+
+func (a *auditLogConfigurator) CanEnableAuditLogsForShoot(seedName string) bool {
+	return seedName != "" && a.auditLogTenantConfigPath != ""
+}
+
+func (a *auditLogConfigurator) SetAuditLogAnnotation(shoot *gardener_types.Shoot, seed gardener_types.Seed) (bool, error) {
+	data, err := a.getConfigFromFile()
```
Comment (franpog859) [inline components/provisioner/internal/gardener/auditlogging.go:36]: What do you think about naming the `data` -> `auditlogConfig` or something more descriptive?

File: components/provisioner/internal/gardener/auditlogging.go (hunk 2 above)
Comment (Maladie) [reply components/provisioner/internal/gardener/auditlogging.go:13]: I did, but couldn't find it

File: components/provisioner/internal/gardener/shoot_reconciler.go (hunk 5 above)
Comment (Maladie) [reply components/provisioner/internal/gardener/shoot_reconciler.go:76]: No idea :D

File: components/provisioner/internal/gardener/auditlogging.go (hunk 2 above)
Comment (franpog859) [reply components/provisioner/internal/gardener/auditlogging.go:13]: Oh, ok 😢

**Review by franpog859 — APPROVED** [review]
LGTM
